lang/cs/Org.Apache.REEF.IMRU.Tests/TestTaskManager.cs

```diff
@@ -0,0 +1,702 @@
+        /// </summary>
+        [Fact]
+        public void TestAddingClosingRunningTasks()
+        {
+            var taskManager = TaskManagerWithTasksAdded();
+            taskManager.SubmitTasks();
+
+            taskManager.RecordRunningTask(CreateMockRunningTask(MasterTaskId));
+            taskManager.RecordRunningTask(CreateMockRunningTask(MapperTaskIdPrefix + 1));
+
+            var runningTask2 = CreateMockRunningTask(MapperTaskIdPrefix + 2);
+            taskManager.RecordRunningTaskDuringSystemFailure(runningTask2, TaskManager.CloseTaskByDriver);
+
+            taskManager.CloseAllRunningTasks(TaskManager.CloseTaskByDriver);
+            Assert.True(taskManager.AreAllTasksInState(TaskState.TaskWaitingForClose));
+        }
+
+        /// <summary>
+        /// Tests set tasks to fail for running tasks
+        /// </summary>
+        [Fact]
+        public void TestFailedRunningTasks()
+        {
+            var taskManager = TaskManagerWithTasksAdded();
+            taskManager.SubmitTasks();
+
+            taskManager.RecordRunningTask(CreateMockRunningTask(MasterTaskId));
+            taskManager.RecordRunningTask(CreateMockRunningTask(MapperTaskIdPrefix + 1));
+            taskManager.RecordRunningTask(CreateMockRunningTask(MapperTaskIdPrefix + 2));
+
+            taskManager.RecordFailedTaskDuringRunningOrSubmissionState(CreateMockFailedTask(MapperTaskIdPrefix + 1, TaskManager.TaskAppError));
+            taskManager.RecordFailedTaskDuringRunningOrSubmissionState(CreateMockFailedTask(MapperTaskIdPrefix + 2, TaskManager.TaskGroupCommunicationError));
+            taskManager.RecordFailedTaskDuringRunningOrSubmissionState(CreateMockFailedTask(MasterTaskId, TaskManager.TaskSystemError));
+            Assert.True(taskManager.AllInFinalState());
+        }
+
+        /// <summary>
+        /// Tests number of application errors 
+        /// </summary>
+        [Fact]
+        public void TestAppError()
+        {
+            var taskManager = TaskManagerWithTasksAdded();
+            taskManager.SubmitTasks();
+            taskManager.RecordFailedTaskDuringRunningOrSubmissionState(CreateMockFailedTask(MapperTaskIdPrefix + 1, TaskManager.TaskAppError));
+            taskManager.RecordFailedTaskDuringRunningOrSubmissionState(CreateMockFailedTask(MapperTaskIdPrefix + 2, TaskManager.TaskSystemError));
+            Assert.Equal(1, taskManager.NumberOfAppError());
+        }
+
+        /// <summary>
+        /// Tests failed tasks in various event sequences
+        /// </summary>
+        [Fact]
+        public void TestFailedTasks()
+        {
+            var taskManager = TaskManagerWithTasksAdded();
+            taskManager.SubmitTasks();
+
+            taskManager.RecordRunningTask(CreateMockRunningTask(MapperTaskIdPrefix + 1));
+            taskManager.RecordRunningTask(CreateMockRunningTask(MapperTaskIdPrefix + 2));
+
+            //// This task failed by evaluator then failed by itself
```

Comment:
I'm not saying to remove comments, sorry for being confusing, I'm saying to use `//`-style for them